Sources/MapboxNavigation/WayNameView.swift

```diff
@@ -2,12 +2,137 @@ import Foundation
 import UIKit
 import Turf
 import MapboxMaps
+import MapboxDirections
 
 /**
  A label that is used to show a road name and a shield icon.
  */
 @objc(MBWayNameLabel)
-open class WayNameLabel: StylableLabel {}
+open class WayNameLabel: StylableLabel {
+    var spriteRepository = SpriteRepository()
+    var representation: VisualInstruction.Component.ImageRepresentation?
+    
+    // When the map style changes, update the sprite repository and the label.
+    func updateStyle(styleURI: StyleURI?) {
+        guard let styleURI = styleURI else { return }
+        spriteRepository.updateRepository(styleURI: styleURI, representation: representation) { [weak self] in
+            guard let self = self else { return }
+            if let roadName = self.text {
+                self.setUpWith(roadName: roadName)
+            }
+        }
+    }
+    
+    func updateRoad(roadName: String, representation: VisualInstruction.Component.ImageRepresentation? = nil) {
+        // When the imageRepresentation of road shield changes, update the sprite repository and the label.
+        if representation != self.representation {
+            spriteRepository.updateRepository(representation: representation) { [weak self] in
+                guard let self = self else { return }
+                self.representation = representation
+                self.setUpWith(roadName: roadName)
+            }
+        }
+        self.representation = representation
+        setUpWith(roadName: roadName)
+    }
+    
+    // Set up the `WayNameLabel` with the road name. Try to use the Mapbox designed shield first, if failed, fall back to use the legacy road shield icon.
+    // If there's no valid shield image, display the road name only.
+    private func setUpWith(roadName: String) {
+        // TODO: Use shield for us-state shield first
+        if let shield = representation?.shield,
+           shield.name != "us-state",
```

Comment:
If the legacy endpoint doesn’t have an image for a particular `us-state` shield, we should fall back to the new codepath – essentially the reverse of the fallback for other shield types.